### ISSUE TITLE
[Enhancement] [RHEL/6] [RHEL/7] PCI-DSS profile -- Provide profile rules mapping to PCI-DSS rule requirements

### ISSUE THREAD

**Issue by iankko** [OpenSCAP/scap-security-guide#606]
The [version 3.0 of the PCI-DSS standard](https://www.pcisecuritystandards.org/documents/PCI_DSS_v3.pdf) defines the following requirements:
* ```Requirement 1: Install and maintain a firewall configuration to protect cardholder data```,

* ```Requirement 2: Do not use vendor-supplied defaults for system passwords and other security parameters```,

* ```Requirement 3: Protect stored cardholder data```,

* ```Requirement 4: Encrypt transmission of cardholder data across open, public networks```,

* ```Requirement 5: Protect all systems against malware and regularly update anti-virus software or programs```,

* ```Requirement 6: Develop and maintain secure systems and applications```,

* ```Requirement 7: Restrict access to cardholder data by business need to know```,

* ```Requirement 8: Identify and authenticate access to system components```,

* ```Requirement 9: Restrict physical access to cardholder data```,

* ```Requirement 10: Track and monitor all access to network resources and cardholder data```,

* ```Requirement 11: Regularly test security systems and processes```, and

* ```Requirement 12: Maintain a policy that addresses information security for all personnel.```

Therefore enhance the existing SSG ```PCI-DSS``` profiles in ```RHEL/6``` and ```RHEL/7``` products to map existing rules to aforementioned PCI-DSS standard requirements.

**Comment by iankko** [comment]
To provide an example how this PCI-DSS v3 Requirement # mapping to SSG RHEL-7's PCI-DSS rules mapping would be performed -- let's consider the case of ```Ensure Software Patches Installed``` for the example below:

The PCI-DSS v3 Requirements and Security Assessments procedures PDF:
&nbsp; &nbsp; https://www.pcisecuritystandards.org/documents/PCI_DSS_v3.pdf

contains (page #46)
 ``` 
Maintain a Vulnerability Management Program

Requirement 5:
Protect all systems against malware and regularly update anti-virus software or programs
```

So we would map PCI-DSS v3 Req #5 to RHEL-7's PCI-DSS ```Ensure Software Patches Installed``` for the beginning (in the future more rules than just only one can shown to map properly on ```Req #5``` but the process would be the same).

Having look at ```Ensure Software Patches Installed``` XCCDF rule definition in RHEL-7 product:
&nbsp; &nbsp; https://github.com/OpenSCAP/scap-security-guide/blob/master/RHEL/7/input/system/software/updating.xml#L113

there's a ```<ref>``` element:
&nbsp; &nbsp; https://github.com/OpenSCAP/scap-security-guide/blob/master/RHEL/7/input/system/software/updating.xml#L138

So we would enhance it from the current form:
```<ref nist="SI-2,MA-1(b)" disa=""/>```
to something like:
```<ref nist="SI-2,MA-1(b)" disa="" pcidss="Req-5"/>```.

The similar action needs to be performed for every rule currently present in RHEL-7's PCI-DSS profile (take each rule, review the XCCDF prose for it, find the corresponding requirement in the PCI-DSS v3 PDF document && enhance the <ref> element for that rule with PCI-DSS Req refence).

This is one part of the task. Though yet before we can start assigning the PCI-DSS Req # IDs, the RHEL-7 reference processing XSLT transformation needs to be modified a bit too.

Having a look e.g. at:
&nbsp; &nbsp; https://github.com/OpenSCAP/scap-security-guide/blob/master/RHEL/7/transforms/shorthand2xccdf.xslt#L158

there's visible XSLT transformation how we are modifying the shorthand XML file, when "disa" attributet in the <ref> element is encountered (we replace that value with the value of "disa-cciuri").

That "disa-cciuri' is defined in ```RHEL/7/transforms/constants.xslt``` file.

So when performing the first PCI-DSS Req # mapping, both the ```RHEL/7/transforms/constants.xslt``` XSLT transformation and ```RHEL/7/transforms/shorthand2xccdf.xslt``` transformation will need to get updated (```constants.xslt``` will get equipped with new "pcidssuri" variable, pointing to 'https://www.pcisecuritystandards.org/documents/PCI_DSS_v3.pdf', and ```shorthand2xccdf.xslt``` transformation will be equipped with similar XSLT block for the case when newly added "pcidss" attribute is encountered within the "<ref>" element of the input shorthand file. In ```shorthand2xccdf.xslt``` transformation we would translate the PCI-DSS Req # ID to URL of that PDF PCI-DSS v3 document).

**Comment by ybznek** [comment]
So, I should
- get selected rules from https://github.com/OpenSCAP/scap-security-guide/blob/master/RHEL/7/input/profiles/pci-dss.xml
- find their definitions of rules^
- read description/title/... of rule and find corresponding "number" in https://www.pcisecuritystandards.org/documents/PCI_DSS_v3.pdf
- modify shorthand2xccdf.xslt  (add pcidss support)
- add element to contants.xslt

**Comment by iankko** [comment]
@ybznek 
> So, I should
> 
    get selected rules from https://github.com/OpenSCAP/scap-security-guide/blob/master/RHEL/7/input/profiles/pci-dss.xml
    find their definitions of rules^
    read description/title/... of rule and find corresponding "number" in https://www.pcisecuritystandards.org/documents/PCI_DSS_v3.pdf

And for each such a rule add new "pcidss=Req # appropriate PCI-DSS ID here" attribute to the ```<ref>``` element for that rule.

Otherwise it's correct.

>    modify shorthand2xccdf.xslt (add pcidss support)
>    add element to contants.xslt

This would be done just once (when first first ```<ref>``` is updated with new "pcidss" attribute").

Otherwise it's correct.

**Comment by iankko** [comment]
Tracking so far unresolved issues here:
> Issue by @ybznek:

I was doing my best, but the mapping of some rules wasn't easy.

Still I have some rules without mapping.

-audit_rules_media_export
- # 9 is about medias, but about their storage/transportation
- # 10 explicitly doesn't mention something with external media

-display_login_attempts
- best match with 10.2.1, but I think it isn't the same.

I have mapped userowner_shadow_file to # 8, because It makes password cracking more complicated, but I don't know how to map these rules:
-file_owner_etc_group
-file_groupowner_etc_group
-file_permissions_etc_group
-file_owner_etc_passwd
-file_groupowner_etc_passwd
-file_permissions_etc_passwd
-file_user_owner_grub2_cfg
-file_group_owner_grub2_cfg

**Comment by iankko** [comment]
Issue from Shawn:
> @shawndwells 

This work is great! Could the make table-refs be updated as part of this?

e.g.
```
table-refs: shorthand2xccdf
        xsltproc -stringparam ref "nist" -o $(OUT)/table-$(PROD)-nistrefs.html $(TRANS)/xccdf2table-byref.xslt $(OUT)/unlinked-$(PROD)-empty-groups-xccdf.xml
        xsltproc -stringparam profile "common" -o $(OUT)/table-$(PROD)-nistrefs-common.html \
                $(TRANS)/xccdf2table-profilenistrefs.xslt $(OUT)/unlinked-$(PROD)-empty-groups-xccdf.xml
.....
        xsltproc -stringparam ref "pcidss" -o $(OUT)/table-$(PROD)-pcidss.html $(TRANS)/xccdf2table-byref.xslt $(OUT)/unlinked-$(PROD)-empty-groups-xccdf.xml
        xsltproc -stringparam profile "pci-dss" -o $(OUT)/table-$(PROD)-pcidss.html \
                $(TRANS)/xccdf2table-profilenistrefs.xslt $(OUT)/unlinked-$(PROD)-empty-groups-xccdf.xml
```

**Comment by iankko** [comment]
The ```table-refs``` action will be covered via the new ticket:
&nbsp; &nbsp; [1]  https://github.com/OpenSCAP/scap-security-guide/issues/801

Since this wasn't part of the original issue, will be corrected in via issue #801. The original requirement (map all PCI-DSS profile rules to PCI-DSS v3 standard requirements is done). Underlying PRs, performing this, are as follows:
&nbsp; &nbsp; [2] https://github.com/OpenSCAP/scap-security-guide/pull/659
&nbsp; &nbsp; [3] https://github.com/OpenSCAP/scap-security-guide/pull/792
